### PR TITLE
open proposal XDV 

### PR DESCRIPTION
**Pull request by molekilla**
V 1.0 proposal done

### CODE REVIEW

**Comment by realChainLife** [comment]
Hi @molekilla thank you for the proposal. Can you rework to reflect budget breakdown per milestone? Also please include your initial documentation plans in your deliverables.

**Comment by molekilla** [comment]
@realChainLife done, had to rework the initial idea, should be more realistic to implement. thanks

File: open-grant-proposals/open-proposal-xdv.md [hunk 1]
```diff
@@ -0,0 +1,378 @@
...
+
+
+## Deliverables
+
+
+### Metadata / File storage rust-ipld
+
+#### Spec
+
+Create a multiformat `xdv` and migrate PoC to XDV Node (Go) and/or Rust. The API must be 100% compatible with IPFS, ie `POST /add` and other relevant REST methods. It must have an interface with events to register `cross chain swaps` events.
```
Comment (vmx) [inline open-grant-proposals/open-proposal-xdv.md:121]: What do you mean by "multiformat `xdv`". Would you need a new multiformat codec? Earlier in the document you mention that you'll use `dag-cbor`.

> The API must be 100% compatible with IPFS, ie `POST /add` and other relevant REST methods.

Is the idea to just be a proxy and use a full IPFS node under the hood, or re-implementing those. Also "relevent REST methods" is pretty vague, the IPFS API is pretty large, which ones did you have in mind? :)

File: open-grant-proposals/open-proposal-xdv.md [hunk 2]
```diff
@@ -0,0 +1,378 @@
...
+
+- What are the risks that will make executing on this project difficult?
+finding knowledgable IPLD engineers
+
+
+
+## Deliverables
+
+
+### Metadata / File storage rust-ipld
```
Comment (vmx) [inline open-grant-proposals/open-proposal-xdv.md:117]: You mention "rust-ipld" here for file storage. Do you plan to implement a full storage solution yourself? `rust-ipld` would only be a small part of that, then encoding/decoding of the dag-cbor data.

File: open-grant-proposals/open-proposal-xdv.md (hunk 1 above)
Comment (molekilla) [reply open-grant-proposals/open-proposal-xdv.md:121]: 1. sorry,  no need to do new codec
2. only Files (add, get)

File: open-grant-proposals/open-proposal-xdv.md (hunk 2 above)
Comment (molekilla) [reply open-grant-proposals/open-proposal-xdv.md:117]: secret network / cosmos handles storage, rust is for  encoding and decoding

**Comment by realChainLife** [comment]
Hi @molekilla thank you for this proposal. We would like to fund the work outlined in this proposal. Can you email devgrants@fil.org so we can discuss next steps.

**Pull request merged**